build/deploy/main.go

```diff
@@ -34,11 +34,29 @@ func deploy() error {
 	bundlePath := os.Args[2]
 
 	siteURL := os.Getenv("MM_SERVICESETTINGS_SITEURL")
+	adminToken := os.Getenv("MM_ADMIN_TOKEN")
 	adminUsername := os.Getenv("MM_ADMIN_USERNAME")
 	adminPassword := os.Getenv("MM_ADMIN_PASSWORD")
 	copyTargetDirectory, _ := filepath.Abs("../mattermost-server")
-	if siteURL != "" && adminUsername != "" && adminPassword != "" {
-		return uploadPlugin(pluginID, bundlePath, siteURL, adminUsername, adminPassword)
+
+	if siteURL != "" {
+		client := model.NewAPIv4Client(siteURL)
+
+		if adminToken != "" {
+			log.Printf("Authenticating using token against %s.", siteURL)
+			client.SetToken(adminToken)
+
+			return uploadPlugin(client, pluginID, bundlePath)
+		} else if adminUsername != "" && adminPassword != "" {
```

Comment:
No need for the `else`

Reply:
This is catching the case where just `siteURL` is defined, but no useful credentials, where we continue to fall back to the copy target directory method. I regularly run with an `MM_SERVICESETTINGS_SITEURL` defined, and could imagine others doing the same, but for whatever reason might still want to use the copying methodology.

That being said, copying is becoming less and less useful. I've already reorganized my meaningful GOPATH projects into directory hierarchies making these directories no longer peers. It might make sense to deprecate this altogether at some point.

Reply:
It agree with all you said, but my comment was not about the code semantic. I was just commenting on the style.

My proposal is to rewrite it as:

```suggestion
		}
		
		if adminUsername != "" && adminPassword != "" {
```

Reply:
Oh, I see! Makes sense :)